packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+            ECCODESL='-L' + self.spec['eccodes'].prefix + '/lib '
+            ECCODES_LIBS='-leccodes '
+            CPPFLAGS+=ECCODESI
+            LDFLAGS+=ECCODESL
+            LIBS+=ECCODES_LIBS
+
+        # Claw library
+        if '+claw' in self.spec:
+            args.append('CLAW=' + self.spec['claw'].prefix + '/bin/clawfc')
+            args.append('CLAWFLAGS=-I' + self.spec['netcdf-c'].prefix + '/' + self.compiler.name + '/' + str(self.compiler.version.up_to(2))  + '/include')
+            args.append('--enable-claw')
+
+        # Rte-rttmgp
+        if '+rte-rrtmgp' in self.spec:
+            args.append('--enable-rte-rrtmgp')
+
+        # Mpi-checks
+        if '~mpi-checks' in self.spec:
+            args.append('--disable-mpi-checks')
+
+        # OpenMP library
+        if '~openmp' in self.spec:
+            args.append('--disable-openmp')
+        else:
+            LIBS+='-lomptarget '
+            LDFLAGS+=' -L' + self.spec['ompt-openmp'].prefix + ' '
+
+        # Serialbox library
+        if '+serialize' in self.spec:
+            SERIALBOXI='-I' + self.spec['serialbox'].prefix + '/include '
+            SERIALBOXL='-L' + self.spec['serialbox'].prefix + '/lib '
+            SERIALBOX_LIBS='-lSerialboxFortran '
+            FCFLAGS+=SERIALBOXI
+            LDFLAGS+=SERIALBOXL
+            LIBS+=SERIALBOX_LIBS
+
+            args.append('SB2PP=python2 ' + self.spec['serialbox'].prefix +  '/python/pp_ser/pp_ser.py ')
+            if self.spec.variants['icon_target'].value == 'gpu':
+                args.append('--enable-serialization=read')
+            else:
+                args.append('--enable-serialization=create')
+
+        # Libxml2 library
+        XML2I='-I' + self.spec['libxml2'].prefix + '/include/libxml2 '
+        XML2L='-L' + self.spec['libxml2'].prefix + '/lib '
+        XML2L_LIBS='-lxml2 '
+
+        # Blas lapack library
+        BLAS_LAPACK_LIBS='-llapack -lblas '
+
+        # Netcdf library
+        if self.compiler.name == 'cce':
+            NETCDFL='-L' + self.spec['netcdf-c'].prefix + '/crayclang/9.0/lib '
+        else:
+            NETCDFL='-L' + self.spec['netcdf-c'].prefix + '/' + self.compiler.name + '/' + str(self.compiler.version.up_to(2))  + '/lib '
+        NETCDF_LIBS='-lnetcdf -lnetcdff '
+
+        # Set MPI_LAUNCH
+        args.append('MPI_LAUNCH=false')
+
+        # CPPFLAGS
+        CPPFLAGS+=XML2I
+
+        # PGI compiler flags
+        if self.compiler.name == 'pgi':
+            # Standard cpp library
+            STDCPPL='-L/opt/gcc/8.3.0/snos/lib64 '
+            STDCPP_LIBS='-lstdc++ '
+
+            # CFLAGS
+            CFLAGS+='-g -O2 '
+
+            # FCFLAGS
+            FCFLAGS+='-g -O -Mrecursive -Mallocatable=03 '
+
+            # PGI GPU
+            if self.spec.variants['icon_target'].value == 'gpu':
+                # NVCFLAGS
+                args.append('NVCC=nvcc')
+                args.append('NVCFLAGS=--std=c++11 -arch=sm_60 -g -O3')
```

Comment:
For the `FCFLAGS` the compute capability is parametrized by `self.spec.variants['cuda_arch'].value`. For consistency, this should be the case here aswell.

Reply:
you're right thx :)